handsontable/src/utils/parseTable.js

```diff
@@ -18,137 +20,586 @@ function isHTMLTable(element) {
   return (element && element.nodeName || '') === 'TABLE';
 }
 
+/**
+ * Parses empty values to an empty string or leave them untouched otherwise.
+ *
+ * @private
+ * @param {string} cellValue Parsed cell value.
+ * @returns {string}
+ */
+function parseEmptyValues(cellValue) {
+  if (isEmpty(cellValue)) {
+    return '';
+  }
+
+  return cellValue;
+}
+
 /**
  * Converts Handsontable into HTMLTableElement.
  *
  * @param {Core} instance The Handsontable instance.
  * @returns {string} OuterHTML of the HTMLTableElement.
  */
 export function instanceToHTML(instance) {
-  const hasColumnHeaders = instance.hasColHeaders();
-  const hasRowHeaders = instance.hasRowHeaders();
-  const coords = [
-    hasColumnHeaders ? -1 : 0,
-    hasRowHeaders ? -1 : 0,
-    instance.countRows() - 1,
-    instance.countCols() - 1,
+  const startColumn = instance.hasRowHeaders() ? -1 : 0;
+  const startRow = instance.hasColHeaders() ? -1 : 0;
+  const rows = Array.from({ length: instance.countRows() + Math.abs(startRow) },
+    (_, i) => i + startRow);
+  const columns = Array.from({ length: instance.countCols() + Math.abs(startColumn) },
+    (_, i) => i + startColumn);
+
+  return getHTMLByCoords(instance, { rows, columns });
+}
+
+/**
+ * Converts Handsontable's coordinates into HTMLTableElement.
+ *
+ * @param {Core} instance The Handsontable instance.
+ * @param {object} config Configuration for building HTMLTableElement.
+ * @param {Array<number>} config.rows List of row indexes which should be taken into account when creating the table.
+ * @param {Array<number>} config.columns List of column indexes which should be taken into account when creating the table.
+ * @returns {string} OuterHTML of the HTMLTableElement.
+ */
+export function getHTMLByCoords(instance, config) {
```

Comment:
```suggestion
export function getHTMLByCoords(hotInstance, config) {
```
I'd highly recommend changing the name of the argument to be more meaningful. Something like `hotInstance` or `hot`.